datahub-graphql-core/src/main/java/com/linkedin/datahub/graphql/resolvers/glossary/CreateGlossaryTermResolver.java

```diff
@@ -100,4 +116,50 @@ private GlossaryTermInfo mapGlossaryTermInfo(final CreateGlossaryEntityInput inp
     }
     return result;
   }
+
+  private Filter buildParentNodeFilter(final Urn parentNodeUrn) {
+    final Map<String, String> criterionMap = new HashMap<>();
+    criterionMap.put(PARENT_NODE_INDEX_FIELD_NAME, parentNodeUrn == null ? null : parentNodeUrn.toString());
+    return QueryUtils.newFilter(criterionMap);
+  }
+
+  private Map<Urn, EntityResponse> getTermsWithSameParent(Urn parentNode, QueryContext context) {
+    try {
+      final Filter filter = buildParentNodeFilter(parentNode);
+      final SearchResult searchResult = _entityClient.filter(
+          Constants.GLOSSARY_TERM_ENTITY_NAME,
+          filter,
+          null,
+          0,
+          1000,
```

Comment:
Does this mean if there were more than 1000 that we might miss some in this validation?

Reply:
that is correct. in reality it's extremely unlikely that anyone has more than 1000 entities inside of one term group / folder... but paginating here might make sense just to guard against unexpected use cases.